frontend/packages/dev-console/src/plugin.tsx

```diff
@@ -353,6 +355,33 @@ const plugin: Plugin<ConsumedExtensions> = [
       getKebabActionsForKind,
     },
   },
+  {
+    type: 'YAMLTemplate',
+    properties: {
+      model: PipelineModel,
+      template: `
+apiVersion: tekton.dev/v1alpha1
+kind: Pipeline
+metadata:
+  name: new-pipeline
```

Comment:
I guess we should use `generateName` or a `Math.Random()` appended to `new-pipeline`

Reply:
I can try that - but I am pretty sure that won't work. The scope of this code is global, so it gets created once and referenced n-times. They'd need to refresh their browser to get a new name.

Reply:
Yup, that won't work. At least it won't let them create two of the same name, so they'll get the prompt to change it. I think this is just a current limitation of the way we deal with default templates. I can look into the YAML editor code to see when it accepts the new block to modify the name, but I am weary about making this wide-reaching change.

Reply:
The scope depends on whether you are storing the template as a variable or returning it through a function. 
Also, I guess there was a keyword like `generateName` introduced in Tekton 0.7. Would need to followup on that.

Reply:
JavaScript scope, not application scope.

This is a static string, it is computed at the browser loading the code. It never gets recomputed, so any "make name" functions run here get executed exactly once... when the file gets cached on the client. You need to refresh to get it to go again.

Reply:
> Also, I guess there was a keyword like generateName introduced in Tekton 0.7. Would need to followup on that.

Are you referring to something that can be put into the YAML? That the server interprets?

Reply:
My bad, I meant while loading it in edit-yaml, line 442. Looking deeper, I guess it might be a big ask and not required at this stage.

Reply:
Right, I agree, I said something to that extent on one of my previous comments.
>  I can look into the YAML editor code to see when it accepts the new block to modify the name, but I am weary about making this wide-reaching change.